addons/mail/static/src/core/web/message_patch.js

```diff
@@ -161,6 +161,8 @@ patch(Message.prototype, {
 
     openRecord() {
         this.message.thread.open({ focus: true });
+        this.store.highlightMessage = this.message;
```

Comment:
We already have `highlightMessage` in the Thread model. We just need to set it to the current message:
```suggestion
        this.message.thread.highlightMessage = this.message;
```
And we only need one line of diff :)

Reply:
I have updated the PR, you can have a look :slightly_smiling_face: